mesonbuild/modules/gnome.py

```diff
@@ -411,8 +411,6 @@ def fix_ldflags(ldflags):
     def _unwrap_gir_target(self, girtarget):
         while hasattr(girtarget, 'held_object'):
             girtarget = girtarget.held_object
-        if not isinstance(girtarget, (build.Executable, build.SharedLibrary)):
```

Comment:
I don't think you should remove this test, but rather add build.StaticLibrary. Otherwise girtarget could be any type.

Reply:
From reading the code, there are only 3 types of targets, or am I missing something?

Reply:
girtarget can be anything, could be a string if you do `gnome.generate_gir('some Strong's, ...)`

Reply:
Done